javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -48,111 +52,192 @@ public static ClientMethodTemplate getInstance() {
         return INSTANCE;
     }
 
-    protected static void addValidations(JavaBlock function, List<String> expressionsToCheck, Map<String, String> validateExpressions, JavaSettings settings) {
-        if (settings.shouldClientSideValidations()) {
-            for (String expressionToCheck : expressionsToCheck) {
-                JavaIfBlock nullCheck = function.ifBlock(expressionToCheck + " == null", ifBlock ->
-                        ifBlock.line("return Mono.error(new IllegalArgumentException(\"Parameter %s is required and "
-                            + "cannot be null.\"));", expressionToCheck));
-                if (validateExpressions.containsKey(expressionToCheck)) {
-                    nullCheck.elseBlock(elseBlock ->
-                            elseBlock.line(validateExpressions.get(expressionToCheck) + ";"));
-                }
+    /**
+     * Adds validations to the client method.
+     *
+     * @param function The client method code block.
+     * @param expressionsToCheck Expressions to validate as non-null.
+     * @param validateExpressions Expressions to validate with a custom validation (key is the expression, value is the
+     * validation).
+     * @param settings AutoRest generation settings, used to determine if validations should be added.
+     */
+    protected static void addValidations(JavaBlock function, List<String> expressionsToCheck,
```

Comment:
unrelated to this PR: Why is this a protected static method?

Reply:
This is THE oldest code that likely converted from C# via some tooling. Could be this reason.

Reply:
Yeah, even spending time writing the Javadocs for these methods and reading through what they do, I'm still not even sure what they do differently.

This would be a good area to begin a minor refactoring when we began to push forward on greenlighting new code generation features.